{% set nav='alleles' %}
{% extends "shared/base.html" %}

{% block title %}Alleles | {{allele | deslugify_allele }}{% endblock %}

{% block breadcrumbs %}
<div class="vertical-spacing-bottom-0-5">
    <small>
        <a href="/alleles">Alleles</a> / 
        <a href="/alleles/species/homo_sapiens/">Human</a> / 
        <a href="/alleles/locus/{{locus}}">{{locus | deslugify_locus}}</a> / 
        <a href="/alleles/allele_group/{{allele_group}}">{{allele_group | deslugify_allele_group | safe}}</a> / 
        <strong>{{allele | deslugify_allele | safe}}</strong>
    </small>
</div>
{% endblock %}


{% block main %}

{% include "styles.html" %}

<section>
    <div class="grid-container">
        <div class="column-full-width">
            <div class="inner">
                <small><strong>Allele</strong></small>
                <h1 class="heading-large vertical-spacing-bottom structure-title">{{allele | deslugify_allele }}</h1>
            </div>
        </div>
    </div>
</section>
<section>
    <div class="grid-container">
        <div class="column-one-third">
            <div class="inner">
                <div>
                    <h3 class="vertical-spacing-top-0-5 vertical-spacing-bottom-0-25 heading-small">Species</h3>
                    <a href="/alleles/species/homo_sapiens">Homo sapiens (Human)</a>
                </div>
                <div>
                    <h3 class="vertical-spacing-top-0-5 vertical-spacing-bottom-0-25 heading-small">Locus</h3>
                    <a href="/alleles/locus/{{locus}}">{{locus | deslugify_locus }}</a>
                </div>
                <div>
                    <h3 class="vertical-spacing-top-0-5 vertical-spacing-bottom-0-25 heading-small">Allele group</h3>
                    <a href="/alleles/allele_group/{{allele_group}}">{{allele_group | deslugify_allele_group }}</a>
                </div>
                <div>
                    <h3 class="vertical-spacing-top-0-5  vertical-spacing-bottom-0-25 heading-small">Canonical allele</h3>
                    <div>
                        <strong>Gene allele:</strong> {{allele_data.canonical_allele.gene_allele_name}}
                    </div>
                    <div>
                        <strong>Identifier:</strong> <a href="https://www.ebi.ac.uk/ipd/imgt/hla/alleles/allele/?accession={{allele_data.canonical_allele.id}}">{{allele_data.canonical_allele.source}}:{{allele_data.canonical_allele.id}}</a>
                    </div>
                </div>
                <div>
                    <h3 class="vertical-spacing-top-0-5  vertical-spacing-bottom-0-25 heading-small">Experimental structures</h3>
                    {% if structure_count > 0 %}<a href="https://www.histo.fyi/structures/browse/alleles/{{allele}}">{% endif %}{{structure_count | structure_count_display}}{% if structure_count > 0 %}{% endif %}</a> 
                </div>
            </div>
        </div>
        <div class="column-two-thirds">
            <div class="inner">
                {% if '-' in allele_data.canonical_sequence %}
                    {% set truncated_sequence = True %}
                    Due to an N-terminal truncation of the sequence of this allele, it does not yet have a structural prediction.
                {% else %}
                    {{polymorphism_view | safe}}
                {% endif %}
            </div>
        </div>
    </div>
</section>  
<section class="vertical-spacing-top-0-5">
    <div class="grid-container">
        <div class="column-full-width">
            <hr />
            <div class="inner">
                <h2 class="vertical-spacing-top-0-5 vertical-spacing-bottom-0-25">Peptide binding pocket polymorphisms</h2> 
                <div class="vertical-spacing-top-0-5 vertical-spacing-bottom-0-5">
                {% if allele != reference_allele %}
                
                    <strong>{{pocket_pseudosequence_match_count}} alleles</strong> have identical binding pocket residues to <strong>{{allele | deslugify_allele }}</strong>. Residues shown in bold are polymorphisms compared to the reference allele, <a href="/alleles/allele/{{reference_allele}}">{{reference_allele | deslugify_allele }}</a>. 
                </div>
                
                <div class="pseudosequence-wrapper">
                    {% for label in pocket_pseudosequence_positions %}<div class="pseudusequence-residue-label">{{label}}</div>{% endfor %}
                </div>
                
                <div class="pseudosequence-wrapper vertical-spacing-bottom-0-5">
                    {% for residue in allele_data.pocket_pseudosequence %}{% set i = loop.index0 %}<div class="pseudusequence-residue sequence a-pocket {{netmhc_pocket_labels[i]}}-pocket {% if pocket_pseudosequence_positions[i] in netmhcpan_polymorphisms %}polymorphic-residue{% else %}non-polymorphic-residue{% endif %}">{{residue}}</div>{% endfor %}
                </div>
                <div>
                    <strong>Pocket key:</strong> <span class="pocket a-pocket">A</span> <span class="pocket b-pocket">B</span> <span class="pocket c-pocket">C</span> <span class="pocket d-pocket">D</span> <span class="pocket e-pocket">E</span> <span class="pocket f-pocket">F</span>
                </div>
                <div class="vertical-spacing-top vertical-spacing-bottom-0-25">
                    The residues listed are ones used commonly for peptide epitope prediction algorithms such as <a href="https://services.healthtech.dtu.dk/services/NetMHCpan-4.1/">NetMHCPan</a> and <a href="{{allele | mhc_flurry_url}}">MHCFlurry</a> and also the <a href="http://mhcmotifatlas.org/">MHCMotifAtlas</a>.
                </div>
                {% else %}
                    <strong>{{allele | deslugify_allele }}</strong> is the reference allele for the {{allele_group | deslugify_allele_group }} allele group, so there are no polymorphisms to show. <strong>{{pocket_pseudosequence_match_count}} alleles</strong> have identical binding pocket residues to <strong>{{allele | deslugify_allele }}</strong>. 
                {% endif %}
            </div>
        </div>
    </div>
</section>
{% set motif = processed_motif %}

{% if motif %}
<section class="vertical-spacing-top-0-5">
    <div class="grid-container">
        <div class="column-full-width">
            <hr />
            <div class="inner">
                <h2 class="vertical-spacing-top-0-5 vertical-spacing-bottom-0-25">Peptide binding motif</h2> 
            </div>
        </div>

        <div class="column-two-thirds">
            <div class="inner">
                <h3 class="vertical-spacing-bottom-0-25"></h3> 
                <div>
                    {#
                    <p>
                        <strong>HLA-B*15:18</strong> has an experimentally determined motif, shown in the logoplot (<strong>a.</strong>). 
                    </p>
                    <p>
                        The motif is shown in the table below (<strong>b.</strong>), with the amino acids at each position shown in order of decreasing frequency. The amino acids shown in bold are the most common at each position.
                    </p>
                    <p>
                        This allele has a strong preference for nonamer (9 amino acid) peptides (<strong>c.</strong>). It has a dominant preference for a Histidine residue at position 2 of the peptide, which is bound by the B-pocket, and a dominant preference for a Tyrosine residue at the C-terminal position, bound by the F-pocket.
                    </p>
                    <p>
                        Data for the motif of <strong>HLA-B*15:18</strong> is available from the <a href="http://mhcmotifatlas.org/">MHCMotifAtlas</a>.
                    </p>
                    #}
                </div>
                
        
            </div>
        </div>
        <div class="column-one-third">
            <div class="inner">     
                <h3 class="vertical-spacing-bottom-0-25"><strong>a.</strong> Nonamer motif logoplot</h3>   
                <img src="https://images.histo.fyi/motifs/logoplots/svg/{{motif_allele}}_9.svg" />
            </div>
        </div>

        <div class="column-two-thirds">
            <div class="inner">                
                <h3 class="vertical-spacing-bottom-0-25"><strong>b.</strong> Nonamer motif table</h3>
                <div class="vertical-spacing-bottom">{% if motif_type == 'experimental' %}Experimentally determined {% else %}Inferred {% endif %}motif for <strong>{{allele | deslugify_allele}}</strong> {% if motif_type == 'infered' %} based on pocket residue sequence identity with <a href="/alleles/allele/{{motif_allele}}">{{motif_allele | deslugify_allele}}</a>{% endif %}.</div>
                <table width="100%">
                    <thead>
                        <tr>
                            {% for position in motif %}
                                <th><strong>P{{position}}</strong></th>
                            {% endfor %}
                        </tr>
                    </thead>
                    <tbody>
                        <tr>
                            {% for position in motif %}
                            <td width="11%">
                                {% for amino_acid in motif[position] %}
                                    {% if amino_acid['percentage'] > 5.0 %}
                                        <div class="motif-row {{amino_acid['grade']}}-frequency"><span class="motif-amino-acid ">{{amino_acid['amino_acid']}}</span> <span class="amino-acid-frequency">{{amino_acid['percentage'] | round(1)}}%</span></div>
                                    {% endif %}
                                {% endfor %}
                            </td>
                            {% endfor %}
                        </tr>
                    </tbody>
                </table>
            </div>
        </div>
        <div class="column-one-third">
            <div class="inner">
            
                <h3 class="vertical-spacing-bottom-0-25"><strong>c.</strong> Length preference</h3>
                <div class="">{% if motif_type == 'experimental' %}Experimentally determined {% else %}Inferred {% endif %}peptide length preference for {{allele | deslugify_allele}}.</div>
                <img src="https://images.histo.fyi/motifs/lengthplots/png/{{motif_allele}}.png" width="100%"/>
                
            </div>
        </div>
    </div>
</section>
{% endif %}

{% if not truncated_sequence %}
<section class="vertical-spacing-top-0-5">
    <div class="grid-container">
        <div class="column-full-width">
            <div class="inner">
                <h2 class="vertical-spacing-top-0-5 vertical-spacing-bottom-0-25">Structure prediction</h2>
                <h3 class="heading-small vertical-spacing-top-0-5">AlphaFold quality metrics for the prediction.</h3>
            </div>
        </div>
        <div class="column-one-half">
            <div class="inner">
                <h3><strong>a.</strong> pLDDt for {{allele | deslugify_allele}} prediction</h3>
                <img src="https://images.histo.fyi/predictions/alphafold/{{locus}}/{{allele}}_plddt.png" />
            </div>
        </div>
        <div class="column-one-half">
            <h3><strong>b.</strong> Sequence coverage for {{allele | deslugify_allele}} prediction</h3>
            <img src="https://images.histo.fyi/predictions/alphafold/{{locus}}/{{allele}}_coverage.png" />
        </div>
        <div class="column-full-width">
            <div class="inner">
                <h3><strong>c.</strong> PAE plots for the 5 ranked predictions for {{allele | deslugify_allele}}</h3>
                <img src="https://images.histo.fyi/predictions/alphafold/{{locus}}/{{allele}}_pae.png" />
            </div>
        </div>
        <div class="column-full-width">
            <div class="inner">
                <h3 class="heading-small vertical-spacing-top">Comparison of the prediction to experimentally determined structures.</h3>
                {% if structure_count == 0 %}
                    There are no structures of <strong>{{allele | deslugify_allele }}</strong> to compare the prediction to.
                {% else %}
                    Comparison statistics and plots
                {% endif %}
            </div>
        </div>
    </div>
</section>    
{% endif %}           

{% if structure_count > 0 %}
{% include "section_start.html" %}

                <h2 class="vertical-spacing-top-0-5 vertical-spacing-bottom-0-25">Movements observed in structures</h2>
                {{'Information on side chain mobility if known' | aside | safe}}
                {{'Information on structure backbone flexibility if known' | aside | safe}}

{% include "section_end.html" %}
{% endif %}

{#
{% if not truncated_sequence %}
{% include "section_start.html" %}

                <h2 class="vertical-spacing-top-0-5 vertical-spacing-bottom-0-25">Sequence</h2>
                {{allele_data.canonical_sequence}}
            
                {{'Sequence with location of polymorphisms' | aside | safe}}

{% include "section_end.html" %}
{% endif %}
#}

{% endblock %}